tests/e2e/kcs/test_monitor_recovery.py

```diff
@@ -0,0 +1,838 @@
+            self.dc_pods.append(
+                dc_pod_factory(
+                    interface=constants.CEPHFILESYSTEM,
+                    pvc=cephfs_pvc,
+                )
+            )
+        dd_cmd = f"dd if=/dev/urandom of=/tmp/{self.filename} bs=4M count=3"
+        run_cmd(cmd=dd_cmd)
+        for pod_obj in self.dc_pods:
+            pod.upload(
+                pod_obj.name,
+                f"/tmp/{self.filename}",
+                "/mnt/",
+                namespace=project.namespace,
+            )
+            # Calculate md5sum
+            md5sum = pod.cal_md5sum(pod_obj, self.filename)
+            pod_obj.pvc.md5sum = md5sum
+        logger.info("IO completed on all pods")
+
+        logger.info("Running ios on OBCs")
+        self.sanity_helpers.obc_put_obj_create_delete(mcg_obj, bucket_factory)
+
+    def test_monitor_recovery(self):
+        """
+        Verifies Monitor recovery procedure as per:
+        https://access.redhat.com/documentation/en-us/red_hat_openshift_container_storage/4.9/
+        html-single/troubleshooting_openshift_container_storage/index
+
+        """
+        # Initialize mon recovery class
+        mon_recovery = MonitorRecovery()
+
+        logger.info("Backing up all the deployments")
+        mon_recovery.backup_deployments()
+        mons_revert = mon_recovery.mon_deployments_to_revert()
+        mds_revert = mon_recovery.mds_deployments_to_revert()
+
+        logger.info("Corrupting ceph monitors by deleting store db")
+        corrupt_ceph_monitors()
+
+        logger.info("Starting the monitor recovery procedure")
+        logger.info("Scaling down rook and ocs operators")
+        mon_recovery.scale_rook_ocs_operators(replica=0)
+
+        logger.info(
+            "Patching OSDs to remove LivenessProbe and setting sleep to infinity"
+        )
+        mon_recovery.patch_sleep_on_osds()
+
+        logger.info("Getting mon-store from OSDs")
+        mon_recovery.get_monstore_from_osds()
+
+        logger.info("Patching MONs to sleep infinitely")
+        mon_recovery.patch_sleep_on_mon()
+
+        logger.info("Updating initial delay on all monitors")
+        update_mon_initial_delay()
+
+        logger.info("Generating monitor map command using the IPs")
+        self.mon_map_cmd = generate_monmap_cmd()
+
+        logger.info("Getting ceph keyring from ocs secrets")
+        self.keyring_files = mon_recovery.get_ceph_keyrings()
+
+        logger.info("Rebuilding Monitors to recover store db")
+        mon_recovery.monitor_rebuild()
+
+        logger.info("Reverting mon, osd and mgr deployments")
+        mon_recovery.revert_patches(mons_revert)
+
+        logger.info("Scaling back rook and ocs operators")
+        mon_recovery.scale_rook_ocs_operators(replica=1)
+
+        logger.info("Sleeping for 150 secs for cluster to stabilize")
+        time.sleep(150)
+        logger.info("Recovering CephFS")
+        mon_recovery.scale_rook_ocs_operators(replica=0)
```

Comment:
why is replica zero?

Reply:
Need to recover cephfs after monitor recovery, so need to scale down rook and ocs operators so that we can patch mds pods to sleep infinitely until the procedure is done.